notebooks/beir_cqadupstack.ipynb

```diff
@@ -0,0 +1,546 @@
+    "If desired or necessary it is possible to limit the number of used queries.\n",
+    "(to speed up the evaluation)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "21048042-693b-44c5-be92-63ba12783ef0",
+   "metadata": {
+    "id": "21048042-693b-44c5-be92-63ba12783ef0",
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from pv211_utils.beir.loader import load_documents,load_queries,load_judgements\n",
+    "\n",
+    "# generally the same set of documents is used for train, dev, and test - no need to store it multiple times\n",
+    "documents = load_documents(raw_corpus_test)\n",
+    "max_test_queries = 50 # for HotpotQA I suggest for example just 200 test queries, None == use all avaialbe \n",
+    "test_queries = load_queries(raw_queries_test, max_test_queries)\n",
+    "test_judgements = load_judgements(test_queries, documents, raw_qrels_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "H6oU2MTN54pF",
+   "metadata": {
+    "id": "H6oU2MTN54pF",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Implement the Information Retrieval system\n",
+    "\n",
+    "Next, we will define a class named `IRSystem` that will represent your information retrieval system. Your class must define a method name `search` that takes a query and returns documents in descending order of relevance to the query.\n",
+    "\n",
+    "This example returns documents in a decreasing order according to\n",
+    "a [*Okapi BestMatch25*](https://en.wikipedia.org/wiki/Okapi_BM25) similarity score between the documents and the given query.\n",
+    "\n",
+    "It also allows for the use of a [*re-ranking*](https://developers.google.com/machine-learning/recommendation/dnn/re-ranking) function, which takes the selected top k results from the base function and reorders them to achieve better results.\n",
+    "\n",
+    "You can use this example as the basis for your own implementation of an Information Retrieval System. Experiment with better suited preprocessing options. Try different base and re-ranking functions. Play around with the various hyperparameters, either by hand or you can use the dev subset to help you find the best setup. And of course, you can scratch this entire piece of code and make something completely new.\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "c30058ea-71b9-4e15-8105-4e427445ab06",
+   "metadata": {
+    "id": "c30058ea-71b9-4e15-8105-4e427445ab06",
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from typing import Iterable\n",
+    "\n",
+    "from multiprocessing import get_context, Pool\n",
+    "from tqdm.notebook import tqdm\n",
+    "\n",
+    "from pv211_utils.irsystem import IRSystemBase \n",
+    "\n",
+    "from gensim.utils import simple_preprocess\n",
+    "from gensim.summarization import bm25\n",
```

Comment:
Replace with functions from `preprocessing/preprocessing.py` and `systems/bm25.py`. Remove `! pip install gensim` on top of the Jupyter notebook.

Reply:
todo

Reply:
should be done